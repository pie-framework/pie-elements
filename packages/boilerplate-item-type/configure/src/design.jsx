import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import cloneDeep from 'lodash/cloneDeep';
import { InputContainer, settings, layout } from '@pie-lib/pie-toolbox/config-ui';
import {EditableHtml} from '@pie-lib/pie-toolbox/editable-html';
import {test, callTest} from 'pie-utils';

callTest('configure');

const { Panel, toggle } = settings;

export class Design extends React.Component {
  static propTypes = {
    model: PropTypes.object.isRequired,
    configuration: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired,
    onModelChanged: PropTypes.func.isRequired,
    onConfigurationChanged: PropTypes.func.isRequired,
    imageSupport: PropTypes.shape({
      add: PropTypes.func.isRequired,
      delete: PropTypes.func.isRequired,
    }),
    uploadSoundSupport: PropTypes.object,
  };

  static defaultProps = {};

  onPromptChanged = (prompt) => {
    const { onModelChanged, model } = this.props;
    const update = cloneDeep(model);

    update.prompt = prompt;
    onModelChanged(update);
  };

  render() {
    const { classes, configuration, imageSupport, model, onConfigurationChanged, onModelChanged, uploadSoundSupport } =
      this.props;
    const {
      contentDimensions = {},
      prompt = {},
      settingsPanelDisabled,
    } = configuration || {};
    const { promptEnabled } = model || {};

    const panelSettings = {};

    const panelProperties = {
      promptEnabled: prompt.settings && toggle(prompt.label),
    };

    return (
      <layout.ConfigLayout
        dimensions={contentDimensions}
        hideSettings={settingsPanelDisabled}
        settings={
          <Panel
            model={model}
            configuration={configuration}
            onChangeModel={(model) => onModelChanged(model)}
            onChangeConfiguration={onConfigurationChanged}
            groups={{
              Settings: panelSettings,
              Properties: panelProperties,
            }}
          />
        }
      >
        {test}
        {promptEnabled && (
          <InputContainer label={prompt.label || ''} className={classes.promptHolder}>
            <EditableHtml
              className={classes.prompt}
              markup={model.prompt}
              onChange={this.onPromptChanged}
              imageSupport={imageSupport}
              uploadSoundSupport={uploadSoundSupport}
              languageCharactersProps={[{ language: 'spanish' }, { language: 'special' }]}
            />
          </InputContainer>
        )}
      </layout.ConfigLayout>
    );
  }
}

export default withStyles((theme) => ({
  promptHolder: {
    width: '100%',
    paddingTop: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2,
  }
}))(Design);
