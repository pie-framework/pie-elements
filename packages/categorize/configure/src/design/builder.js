import cloneDeep from 'lodash/cloneDeep';

export const buildCategories = (categories, choices, correctResponse) => {
  const clonedCategories = cloneDeep(categories);

  return clonedCategories.map(category => {
    const cr = correctResponse.find(cr => cr.category === category.id);

    if (cr) {
      category.choices = (cr.choices || []).map(choiceId => {
        const choice = choices.find(h => h.id === choiceId);
        if (choice) {
          return Object.assign({}, { id: choice.id, content: choice.content });
        }
      });
    }

    return category;
  });
};

const getChoices = (cat, choices, index) => {
  if (!cat.alternateResponses || cat.alternateResponses.length === 0) {
    return [];
  }

  return (cat.alternateResponses[index] || []).map(alt => {
    return choices.find(ch => ch.id === alt);
  });
};

export const buildAlternateResponses = (categories, choices, correctResponse) => {
  const mostAlternates = correctResponse.reduce((mostAlt, cat) => {
    if (cat.alternateResponses && cat.alternateResponses.length >= mostAlt) {
      return cat.alternateResponses.length;
    }

    return mostAlt;
  }, 0);
  const alternatesArray = (new Array(mostAlternates)).fill(0);

  return alternatesArray.map((val, index) => {
    return correctResponse.map((cat, catIndex) => {
      const currentCategory = categories[catIndex];

      return {
        id: currentCategory.id,
        label: currentCategory.label,
        choices: getChoices(cat, choices, index)
      };
    });
  });
};
