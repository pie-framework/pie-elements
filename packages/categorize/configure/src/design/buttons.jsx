import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import Button from '@material-ui/core/Button';
import MuiDivider from '@material-ui/core/Divider';

export class RawAddButton extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    label: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool
  };

  static defaultProps = {
    label: 'Add'
  };

  render() {
    const { classes, className, label, onClick, disabled } = this.props;
    return (
      <Button
        onClick={onClick}
        disabled={disabled}
        className={classNames(classes.addButton, className)}
        size="small"
        variant="contained"
        color="primary"
      >
        {label}
      </Button>
    );
  }
}
const styles = theme => ({
  addButton: {
    height: theme.spacing.unit * 4
  }
});

const AddButton = withStyles(styles)(RawAddButton);

const DeleteButton = withStyles(theme => ({
  deleteButton: {
    margin: 0,
    padding: 0
  }
}))(({ classes, label, onClick }) => (
  <Button
    className={classes.deleteButton}
    onClick={onClick}
    size="small"
    color="primary"
  >
    {label}
  </Button>
));

const Divider = withStyles(theme => ({
  divider: {
    marginTop: 2 * theme.spacing.unit,
    marginBottom: 2 * theme.spacing.unit
  }
}))(({ classes }) => <MuiDivider className={classes.divider} />);

export { AddButton, DeleteButton, Divider };
