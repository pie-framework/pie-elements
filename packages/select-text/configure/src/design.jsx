import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import { withStyles } from '@material-ui/core/styles';
import cloneDeep from 'lodash/cloneDeep';
import { Tokenizer } from '@pie-lib/text-select';
import {
  InputContainer,
  NumberTextField,
  FeedbackConfig,
  settings,
  layout
} from '@pie-lib/config-ui';
import Chip from '@material-ui/core/Chip';
import debug from 'debug';
import EditableHtml from '@pie-lib/editable-html';
import PartialScoring from '@pie-lib/scoring-config';

const { Panel, toggle, radio } = settings;

const log = debug('@pie-element:select-text:configure');

export class Design extends React.Component {
  static propTypes = {
    model: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    className: PropTypes.string,
    onPromptChanged: PropTypes.func.isRequired,
    imageSupport: PropTypes.shape({
      add: PropTypes.func.isRequired,
      delete: PropTypes.func.isRequired
    })
  };

  static defaultProps = {};

  changeText = event => {
    this.apply(u => {
      u.text = event.target.value;
      u.tokens = [];
    });
  };

  changeTokens = (tokens, mode) => {
    this.apply(u => {
      u.tokens = tokens;
      u.mode = mode;

      const correctTokenCount = tokens.filter(t => t.correct).length;
      const max = isFinite(u.maxSelections) ? u.maxSelections : 0;

      u.maxSelections = Math.max(max, correctTokenCount);
    });
  };

  changeMaxSelections = (event, max) => {
    this.apply(u => (u.maxSelections = max));
  };

  apply = fn => {
    const { onChange, model } = this.props;
    const update = cloneDeep(model);
    fn(update);
    onChange(update);
  };

  changeFeedback = feedback => {
    this.apply(u => (u.feedback = feedback));
  };

  changePartialScoring = partialScoring => {
    const { onChange, model } = this.props;
    const update = cloneDeep(model);
    update.partialScoring = partialScoring;
    onChange(update);
  };

  render() {
    const { model, classes, className, onPromptChanged, imageSupport, onChange } = this.props;
    const { configure : {
      prompt,
      text,
      tokens,
      mode,
      feedback,
      partialScoring,
      selections,
      selectionCount,

      correctAnswer,

      teacherInstructions = {},
      studentInstructions = {},
      rationale = {},
      scoringType = {},
      highlightChoices = {}
    }} = model;

    log('[render] maxSelections:', model.maxSelections);


    return (
      <layout.ConfigLayout
        settings={
          <Panel
            model={model}
            onChangeModel={model => onChange(model)}
            groups={{
              'Item Type': {
                highlightChoices: highlightChoices.settings &&
                toggle(highlightChoices.label),
              },
              'Properties': {
                'configure.teacherInstructions.enabled': teacherInstructions.settings &&
                toggle(teacherInstructions.label),
                'configure.studentInstructions.enabled': studentInstructions.settings &&
                toggle(studentInstructions.label),
                'configure.rationale.enabled': rationale.settings &&
                toggle(rationale.label),
                scoringType: scoringType.settings &&
                radio(scoringType.label, ['auto', 'rubric']),
              },
            }}
          />
        }
      >
        <div className={className}>
          <InputContainer label={prompt.label || ''} className={classes.promptHolder}>
            <EditableHtml
              className={classes.prompt}
              markup={model.prompt}
              onChange={onPromptChanged}
              imageSupport={imageSupport}
            />
          </InputContainer>

          {
            text.settings &&
            <TextField
              label={text.label}
              className={classes.input}
              multiline
              value={model.text}
              onChange={this.changeText}
            />
          }

          {
            tokens.settings &&
            <InputContainer label={tokens.label || ''} className={classes.tokenizerContainer}>
              <Tokenizer
                className={classes.tokenizer}
                text={model.text}
                tokens={model.tokens}
                onChange={this.changeTokens}
              />
            </InputContainer>
          }

          {
            mode.settings &&
            <Chip
              label={`${mode.label}: ${model.mode ? model.mode : 'None'}`}
              className={classes.chip}
            />
          }

          {
            selections.settings &&
            <Chip
              label={`${selections.label}: ${model.tokens.length}`}
              className={classes.chip}
            />
          }

          {
            correctAnswer.settings &&
            <Chip
              label={`${correctAnswer.label}: ${
                model.tokens.filter(t => t.correct).length
                }`}
              className={classes.chip}
            />
          }

          {
            selectionCount.settings &&
            <NumberTextField
              min={0}
              label={`${selectionCount.label} (0:any)`}
              max={model.tokens.length}
              value={model.maxSelections}
              onChange={this.changeMaxSelections}
              className={classes.numberField}
            />
          }

          {
            partialScoring.settings &&
            <PartialScoring
              title={'Scoring'}
              label={partialScoring.label}
              partialScoring={model.partialScoring}
              onChange={this.changePartialScoring}
            />
          }
          {
            feedback.settings &&
            <FeedbackConfig
              feedback={model.feedback}
              onChange={this.changeFeedback}
            />
          }

        </div>
      </layout.ConfigLayout>
    );
  }
}
export default withStyles(theme => ({
  tokenizerContainer: {
    paddingRight: 0,
    marginRight: 0,
    '&:after': {
      position: 'absolute',
      bottom: 0,
      left: 0,
      right: 0,
      height: '1px',
      content: '""',
      backgroundColor: theme.palette.primary.main
    },
    marginBottom: theme.spacing.unit
  },
  chip: {
    marginRight: theme.spacing.unit * 2
  },
  input: {
    width: '100%',
    paddingBottom: theme.spacing.unit * 3
  },
  tokenizer: {
    marginTop: theme.spacing.unit * 2
  },
  mainOpts: {
    width: '100%',
    justifyContent: 'space-between',
    display: 'flex',
    alignItems: 'baseline'
  },
  promptHolder: {
    width: '100%',
    paddingBottom: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2
  },
  prompt: {
    paddingTop: theme.spacing.unit * 2,
    width: '100%'
  },
  numberField: {
    width: '180px'
  }
}))(Design);
