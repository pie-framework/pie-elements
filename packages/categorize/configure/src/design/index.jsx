import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import Typography from '@material-ui/core/Typography';
import {
  FeedbackConfig,
  InputContainer,
  layout,
  settings,
} from '@pie-lib/config-ui';
import {
  countInAnswer,
  ensureNoExtraChoicesInAnswer
} from '@pie-lib/categorize';
import EditableHtml from '@pie-lib/editable-html';
import { uid, withDragContext } from '@pie-lib/drag';
import debug from 'debug';

import Categories from './categories';
import AlternateResponses from './categories/alternateResponses';
import Choices from './choices';
import { Divider } from './buttons';
import {
  buildAlternateResponses,
  buildCategories
} from './builder';
import Header from './header';

const { Panel, toggle, radio } = settings;
const { Provider: IdProvider } = uid;

export class Design extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    model: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    uid: PropTypes.string,
    imageSupport: PropTypes.shape({
      add: PropTypes.func.isRequired,
      delete: PropTypes.func.isRequired
    })
  };

  constructor(props) {
    super(props);
    this.uid = props.uid || uid.generateId();
  }

  updateModel = props => {
    const { model, onChange } = this.props;

    const updatedModel = {
      ...model,
      ...props
    };

    //Ensure that there are no extra choices in correctResponse, if the user has decided that only one choice may be used.
    updatedModel.correctResponse = ensureNoExtraChoicesInAnswer(
      updatedModel.correctResponse || [],
      updatedModel.choices
    );

    //clean categories
    updatedModel.categories = updatedModel.categories.map(c => ({
      id: c.id,
      label: c.label
    }));

    updatedModel.choices = updatedModel.choices.map(h => ({
      id: h.id,
      content: h.content,
      categoryCount: h.categoryCount
    }));

    onChange(updatedModel);
  };

  changeRationale = rationale => {
    const { model, onChange } = this.props;

    onChange({
      ...model,
      rationale
    });
  };

  changeTeacherInstructions = teacherInstructions => {
    const { model, onChange } = this.props;

    onChange({
      ...model,
      teacherInstructions
    });
  };

  changeFeedback = feedback => {
    this.updateModel({ feedback });
  };

  onAddAlternateResponse = () => {
    const { model: { correctResponse } } = this.props;

    this.updateModel({
      correctResponse: (correctResponse || []).map(cr => ({
        ...cr,
        alternateResponses: [
          ...(cr.alternateResponses || []),
          []
        ]
      }))
    });
  };

  onPromptChanged = prompt => this.updateModel({ prompt });

  onRemoveAlternateResponse = (index) => {
    const { model: { correctResponse } } = this.props;

    this.updateModel({
      correctResponse: (correctResponse || []).map((cr) => ({
        ...cr,
        alternateResponses: (cr.alternateResponses || []).filter((alt, altIndex) => altIndex !== index)
      }))
    });
  };

  countChoiceInCorrectResponse = choice => {
    const { model } = this.props;
    const out = countInAnswer(choice.id, model.correctResponse);
    return out;
  };

  render() {
    const { classes, className, model, imageSupport, configuration, onChange, onConfigurationChanged } = this.props;
    const {
      partialScoring = {},
      lockChoiceOrder = {},
      teacherInstructions = {},
      studentInstructions = {},
      rationale = {},
      scoringType = {},
      feedback = {},
      prompt = {}
    } = configuration || {};

    const config = model.config || {};
    config.choices = config.choices || { label: '', columns: 2 };

    const categories = buildCategories(
      model.categories || [],
      model.choices || [],
      model.correctResponse || []
    );

    const alternateResponses = buildAlternateResponses(
      model.categories || [],
      model.choices || [],
      model.correctResponse || []
    );

    const choices = model.choices.map(c => {
      c.correctResponseCount = this.countChoiceInCorrectResponse(c);
      return c;
    });

    return (
      <IdProvider value={this.uid}>
        <layout.ConfigLayout
          settings={
            <Panel
              model={model}
              onChangeModel={onChange}
              configuration={configuration}
              onChangeConfiguration={onConfigurationChanged}
              groups={{
                'Settings': {
                  partialScoring: partialScoring.settings &&
                    toggle(partialScoring.label),
                  lockChoiceOrder: lockChoiceOrder.settings &&
                  toggle(lockChoiceOrder.label),
                  'feedback.enabled': feedback.settings && toggle(feedback.label, true),
                },
                'Properties': {
                  'teacherInstructions.enabled': teacherInstructions.settings &&
                    toggle(teacherInstructions.label, true),
                  'studentInstructions.enabled': studentInstructions.settings &&
                    toggle(studentInstructions.label, true),
                  'rationale.enabled': rationale.settings &&
                    toggle(rationale.label, true),
                  scoringType: scoringType.settings &&
                    radio(scoringType.label, ['auto', 'rubric']),
                },
              }}
            />
          }
        >
          <div className={classNames(classes.design, className)}>

            {prompt.settings && (
              <InputContainer
                label={prompt.label}
                className={classes.promptHolder}
              >
                <EditableHtml
                  className={classes.prompt}
                  markup={model.prompt}
                  onChange={this.onPromptChanged}
                  imageSupport={imageSupport}
                  nonEmpty={!prompt.settings}
                  disableUnderline
                />
              </InputContainer>
            )}

            {teacherInstructions.enabled && (
              <InputContainer label={teacherInstructions.label} className={classes.inputHolder}>
                <EditableHtml
                  className={classes.input}
                  markup={model.teacherInstructions || ''}
                  onChange={this.changeTeacherInstructions}
                  imageSupport={imageSupport}
                  nonEmpty={false}
                />
              </InputContainer>
            )}

            {rationale.enabled && (
              <InputContainer label={rationale.label} className={classes.inputHolder}>
                <EditableHtml
                  className={classes.input}
                  markup={model.rationale || ''}
                  onChange={this.changeRationale}
                  imageSupport={imageSupport}
                  nonEmpty={false}
                />
              </InputContainer>
            )}

            <Categories
              imageSupport={imageSupport}
              model={model}
              categories={categories}
              onModelChanged={this.updateModel}
            />

            <Header
              className={classes.alternatesHeader}
              label="Alternate Responses"
              buttonLabel="ADD AN ALTERNATE RESPONSE"
              onAdd={this.onAddAlternateResponse}
            />

            {
              alternateResponses.map((categoriesList, index) => {
                return (
                  <React.Fragment key={index}>
                    <Header
                      className={classes.alternatesHeader}
                      label="Alternate Response"
                      buttonLabel="REMOVE ALTERNATE RESPONSE"
                      onAdd={() => this.onRemoveAlternateResponse(index)}
                    />

                    <AlternateResponses
                      altIndex={index}
                      imageSupport={imageSupport}
                      model={model}
                      categories={categoriesList}
                      onModelChanged={this.updateModel}
                    />

                  </React.Fragment>
                )
              })
            }

            <Divider />
            <Choices
              imageSupport={imageSupport}
              choices={choices}
              model={model}
              onModelChanged={this.updateModel}
            />

            {
              feedback.enabled && (
                <FeedbackConfig
                  feedback={model.feedback}
                  onChange={this.changeFeedback}
                />
              )
            }
          </div>
        </layout.ConfigLayout>
      </IdProvider>
    );
  }
}

const styles = theme => ({
  alternatesHeader: {
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  },
  text: {
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  },
  design: {
    paddingTop: theme.spacing.unit,
    paddingBottom: theme.spacing.unit
  },
  inputHolder: {
    width: '100%',
    paddingBottom: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2
  },
  input: {
    paddingTop: theme.spacing.unit * 2,
    width: '100%',
    maxWidth: '600px'
  },
  prompt: {
    paddingTop: theme.spacing.unit * 2,
    width: '100%'
  },
  promptHolder: {
    width: '100%',
    paddingBottom: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2
  },
  title: {
    marginBottom: '30px'
  }
});

export default withDragContext(withStyles(styles)(Design));
