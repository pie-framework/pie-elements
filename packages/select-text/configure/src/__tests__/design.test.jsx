import { shallow } from 'enzyme';
import React from 'react';
import { Design } from '../design';
import defaultValues from '../defaultConfiguration';

jest.mock('@pie-lib/config-ui', () => ({
  layout: {
    ConfigLayout: props => <div>{props.children}</div>
  },
  settings: {
    Panel: props => <div oChangeModel={props.onChange} />,
    toggle: jest.fn(),
    radio: jest.fn()
  }
}));


describe('design', () => {
  let w;
  let onChange;
  let onPromptChanged;
  let onRationaleChanged;

  const getModel = () => ({
    tokens: [],
  });
  beforeEach(() => {
    onChange = jest.fn();
    onPromptChanged = jest.fn();
    onRationaleChanged = jest.fn();
    w = shallow(
      <Design
        model={getModel()}
        configuration={defaultValues.configuration}
        classes={{}}
        className={'foo'}
        onModelChanged={onChange}
        onPromptChanged={onPromptChanged}
        onRationaleChanged={onRationaleChanged}
      />
    );
  });

  describe('snapshot', () => {
    it('renders all items with defaultProps', () => {
      expect(w).toMatchSnapshot();
    });

    it('renders all items except feedback', () => {
      const defaultConfiguration = defaultValues.configuration;

      defaultConfiguration.feedback.settings = false;

      const wrapper = shallow(
        <Design
          model={getModel()}
          configuration={defaultConfiguration}
          classes={{}}
          className={'foo'}
          onModelChanged={onChange}
          onPromptChanged={onPromptChanged}
          onRationaleChanged={onRationaleChanged}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('renders all items except the content input', () => {
      const defaultConfiguration = defaultValues.configuration;

      defaultConfiguration.text.settings = false;

      const wrapper = shallow(
        <Design
          model={getModel()}
          configuration={defaultConfiguration}
          classes={{}}
          className={'foo'}
          onModelChanged={onChange}
          onPromptChanged={onPromptChanged}
          onRationaleChanged={onRationaleChanged}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    const assert = (fn, args, expected) => {
      const e = expected(getModel());
      it(`${fn} ${JSON.stringify(args)} => ${JSON.stringify(e)}`, () => {
        const instance = w.instance();
        instance[fn].apply(instance, args);
        expect(onChange).toBeCalledWith(e);
      });
    };

    describe('changeText', () => {
      assert('changeText', [{ target: { value: 'foo' } }], m => ({
        ...m,
        text: 'foo'
      }));
    });

    describe('changeTokens', () => {
      assert(
        'changeTokens',
        [[{ start: 0, end: 1, text: 'f' }], 'words'],
        m => ({
          ...m,
          maxSelections: 0,
          tokens: [{ start: 0, end: 1, text: 'f' }],
          mode: 'words'
        })
      );
      assert(
        'changeTokens',
        [[{ start: 0, end: 8, text: 'Foo bar.', correct: true }], 'sentences'],
        m => ({
          ...m,
          maxSelections: 1,
          tokens: [{ start: 0, end: 8, text: 'Foo bar.', correct: true }],
          mode: 'sentences'
        })
      );
    });

    describe('changeMaxSelections', () => {
      assert('changeMaxSelections', [{}, 4], m => ({ ...m, maxSelections: 4 }));
    });

    describe('changeFeedback', () => {
      assert('changeFeedback', [{ correctFeedbackType: 'none' }], m => ({
        ...m,
        feedback: { correctFeedbackType: 'none' }
      }));
    });

    describe('changePartialScoring', () => {
      assert('changePartialScoring', [true], m => ({
        ...m,
        partialScoring: true
      }));
    });

    describe('changePrompt', () => {
      assert('onPromptChanged', ['New Prompt'], m => ({
        ...m,
        prompt: 'New Prompt'
      }));
    });

    describe('changeRationale', () => {
      assert('onRationaleChanged', ['New Rationale'], m => ({
        ...m,
        rationale: 'New Rationale'
      }));
    });


  });
});
