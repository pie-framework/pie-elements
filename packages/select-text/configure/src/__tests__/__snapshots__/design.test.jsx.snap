// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design snapshot renders all items except feedback 1`] = `
<ConfigLayout
  settings={
    <Panel
      configuration={
        Object {
          "correctAnswer": Object {
            "label": "Correct Answers",
            "settings": true,
          },
          "feedback": Object {
            "settings": false,
          },
          "highlightChoices": Object {
            "label": "Highlight choices",
            "settings": true,
          },
          "mode": Object {
            "label": "Mode",
            "settings": true,
          },
          "partialScoring": Object {
            "label": "Allow Partial Scoring",
            "settings": true,
          },
          "prompt": Object {
            "label": "Prompt",
          },
          "rationale": Object {
            "enabled": false,
            "label": "Rationale",
            "settings": true,
          },
          "scoringType": Object {
            "label": "Scoring Type",
            "settings": false,
          },
          "selectionCount": Object {
            "label": "Selection count",
            "settings": true,
          },
          "selections": Object {
            "label": "Selections Available",
            "settings": true,
          },
          "studentInstructions": Object {
            "enabled": true,
            "label": "Student Instructions",
            "settings": false,
          },
          "teacherInstructions": Object {
            "enabled": false,
            "label": "Teacher Instructions",
            "settings": true,
          },
          "text": Object {
            "label": "Content",
            "settings": true,
          },
          "tokens": Object {
            "label": "Tokens",
            "settings": true,
          },
        }
      }
      groups={
        Object {
          "Properties": Object {
            "rationale.enabled": undefined,
            "scoringType": false,
            "studentInstructions.enabled": false,
            "teacherInstructions.enabled": undefined,
          },
          "Settings": Object {
            "highlightChoices": undefined,
          },
        }
      }
      model={
        Object {
          "tokens": Array [],
        }
      }
      onChangeConfiguration={[Function]}
      onChangeModel={[Function]}
    />
  }
>
  <div>
    <Component
      label="Prompt"
    >
      <EditableHtml
        onChange={[Function]}
      />
    </Component>
    <TextField
      label="Content"
      multiline={true}
      onChange={[Function]}
      required={false}
      select={false}
      variant="standard"
    />
    <Component
      label="Tokens"
    >
      <WithStyles(Tokenizer)
        onChange={[Function]}
        tokens={Array []}
      />
    </Component>
    <WithStyles(Chip)
      label="Mode: None"
    />
    <WithStyles(Chip)
      label="Selections Available: 0"
    />
    <WithStyles(Chip)
      label="Correct Answers: 0"
    />
    <Component
      label="Selection count (0:any)"
      max={0}
      min={0}
      onChange={[Function]}
    />
    <Component
      label="Allow Partial Scoring"
      onChange={[Function]}
      title="Scoring"
    />
  </div>
</ConfigLayout>
`;

exports[`design snapshot renders all items except the content input 1`] = `
<ConfigLayout
  settings={
    <Panel
      configuration={
        Object {
          "correctAnswer": Object {
            "label": "Correct Answers",
            "settings": true,
          },
          "feedback": Object {
            "settings": false,
          },
          "highlightChoices": Object {
            "label": "Highlight choices",
            "settings": true,
          },
          "mode": Object {
            "label": "Mode",
            "settings": true,
          },
          "partialScoring": Object {
            "label": "Allow Partial Scoring",
            "settings": true,
          },
          "prompt": Object {
            "label": "Prompt",
          },
          "rationale": Object {
            "enabled": false,
            "label": "Rationale",
            "settings": true,
          },
          "scoringType": Object {
            "label": "Scoring Type",
            "settings": false,
          },
          "selectionCount": Object {
            "label": "Selection count",
            "settings": true,
          },
          "selections": Object {
            "label": "Selections Available",
            "settings": true,
          },
          "studentInstructions": Object {
            "enabled": true,
            "label": "Student Instructions",
            "settings": false,
          },
          "teacherInstructions": Object {
            "enabled": false,
            "label": "Teacher Instructions",
            "settings": true,
          },
          "text": Object {
            "label": "Content",
            "settings": false,
          },
          "tokens": Object {
            "label": "Tokens",
            "settings": true,
          },
        }
      }
      groups={
        Object {
          "Properties": Object {
            "rationale.enabled": undefined,
            "scoringType": false,
            "studentInstructions.enabled": false,
            "teacherInstructions.enabled": undefined,
          },
          "Settings": Object {
            "highlightChoices": undefined,
          },
        }
      }
      model={
        Object {
          "tokens": Array [],
        }
      }
      onChangeConfiguration={[Function]}
      onChangeModel={[Function]}
    />
  }
>
  <div>
    <Component
      label="Prompt"
    >
      <EditableHtml
        onChange={[Function]}
      />
    </Component>
    <Component
      label="Tokens"
    >
      <WithStyles(Tokenizer)
        onChange={[Function]}
        tokens={Array []}
      />
    </Component>
    <WithStyles(Chip)
      label="Mode: None"
    />
    <WithStyles(Chip)
      label="Selections Available: 0"
    />
    <WithStyles(Chip)
      label="Correct Answers: 0"
    />
    <Component
      label="Selection count (0:any)"
      max={0}
      min={0}
      onChange={[Function]}
    />
    <Component
      label="Allow Partial Scoring"
      onChange={[Function]}
      title="Scoring"
    />
  </div>
</ConfigLayout>
`;

exports[`design snapshot renders all items with defaultProps 1`] = `
<ConfigLayout
  settings={
    <Panel
      configuration={
        Object {
          "correctAnswer": Object {
            "label": "Correct Answers",
            "settings": true,
          },
          "feedback": Object {
            "settings": true,
          },
          "highlightChoices": Object {
            "label": "Highlight choices",
            "settings": true,
          },
          "mode": Object {
            "label": "Mode",
            "settings": true,
          },
          "partialScoring": Object {
            "label": "Allow Partial Scoring",
            "settings": true,
          },
          "prompt": Object {
            "label": "Prompt",
          },
          "rationale": Object {
            "enabled": false,
            "label": "Rationale",
            "settings": true,
          },
          "scoringType": Object {
            "label": "Scoring Type",
            "settings": false,
          },
          "selectionCount": Object {
            "label": "Selection count",
            "settings": true,
          },
          "selections": Object {
            "label": "Selections Available",
            "settings": true,
          },
          "studentInstructions": Object {
            "enabled": true,
            "label": "Student Instructions",
            "settings": false,
          },
          "teacherInstructions": Object {
            "enabled": false,
            "label": "Teacher Instructions",
            "settings": true,
          },
          "text": Object {
            "label": "Content",
            "settings": true,
          },
          "tokens": Object {
            "label": "Tokens",
            "settings": true,
          },
        }
      }
      groups={
        Object {
          "Properties": Object {
            "rationale.enabled": undefined,
            "scoringType": false,
            "studentInstructions.enabled": false,
            "teacherInstructions.enabled": undefined,
          },
          "Settings": Object {
            "highlightChoices": undefined,
          },
        }
      }
      model={
        Object {
          "tokens": Array [],
        }
      }
      onChangeConfiguration={[Function]}
      onChangeModel={[Function]}
    />
  }
>
  <div>
    <Component
      label="Prompt"
    >
      <EditableHtml
        onChange={[Function]}
      />
    </Component>
    <TextField
      label="Content"
      multiline={true}
      onChange={[Function]}
      required={false}
      select={false}
      variant="standard"
    />
    <Component
      label="Tokens"
    >
      <WithStyles(Tokenizer)
        onChange={[Function]}
        tokens={Array []}
      />
    </Component>
    <WithStyles(Chip)
      label="Mode: None"
    />
    <WithStyles(Chip)
      label="Selections Available: 0"
    />
    <WithStyles(Chip)
      label="Correct Answers: 0"
    />
    <Component
      label="Selection count (0:any)"
      max={0}
      min={0}
      onChange={[Function]}
    />
    <Component
      label="Allow Partial Scoring"
      onChange={[Function]}
      title="Scoring"
    />
    <Component
      onChange={[Function]}
    />
  </div>
</ConfigLayout>
`;
